dd-sdk-android/src/main/kotlin/com/datadog/android/tracing/internal/net/TracingRequestInterceptor.kt

```diff
@@ -83,7 +87,7 @@ internal class TracingRequestInterceptor(acceptedHosts: List<String> = emptyList
                 // limit spam on 404 resources
                 (span as? MutableSpan)?.resourceName = "404"
             }
-            span.finish()
+            tracedRequestListener.onRequestIntercepted(request, span).finish()
```

Comment:
I think this would be cool to also pass the `response: Response` to the user, as they may want to look into the payload and extract additional info for `span.setTag(...)`. Same for `throwable: Throwable`, which would mean two methods on the listener interface. WDYT 🙂?